cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/MetadataStore.java

```diff
@@ -203,4 +203,9 @@
    * @return the snapshot of the metadata for entities on or before the given time
    */
   Set<MetadataRecord> getSnapshotBeforeTime(MetadataScope scope,  Set<Id.NamespacedId> entityIds, long timeMillis);
+
+  /**
+   * Upgrades the {@link MetadataStore}
+   */
+  void upgrade();
```

Comment:
never mind about the last statement, but still wondering why we can't just keep this at the dataset level